21.md

```diff
@@ -0,0 +1,51 @@
+LUD-21: pinLimit for withdrawRequest
+====================================
+
+`author: titusz` `discussion: https://t.me/lnurl/34810`
+
+---
+
+## Optional Second Factor for `withdrawRequest` Authorization
+
+Adding the optional `pinLimit` property to an LNURL-withdraw response allows a `SERVICE` to require a PIN to authorize a withdrawal above a given amount.
+
+```diff
+{
+    "tag": "withdrawRequest",
+    "callback": string,
+    "defaultDescription": string, 
+    "minWithdrawable": number,  
+    "maxWithdrawable": number,
++   "pinLimit": number
+}
+```
+
+The `pinLimit` value must be a positive integer (including 0) with a maximum of 15 digits. If the `pinLimit` property is present and a `WALLET` (Point of Sale) intends to withdraw an amount equal to or greater than the `pinLimit` value (in millisatoshis) it must first acquire a PIN from the user (customer) and then add it as `pin=<pin>` to the query string of the callback URL to authorize the withdrawal.
```

Comment:
Hm this got me thinking. Let´s take this example:

```
{
    ...
    "minWithdrawable": 1
    "maxWithdrawable": 1000
    "pinLimit": 1001
}
```
This would actually have better backward compatibility because the semantics for minWithdrawable/maxWithdrawable would be retained. If the pinLimit is between, it would **change** the original semantics of `maxWithdrawable` above the pinLimit.

On the other hand - in practice `maxWithdrawable` is often set to a higher value than actually redeemable to prevent clearing out the account.

Any thoughts on this?

Reply:
`maxWithdrawable` on a reusable NFC card (aka bolt card) ends up being set higher than the amount that the system will allow, for two reasons:

1. so that a wallet accidentally seeing the lnurlw as a one-time use gift card/QR will not also accidentally pull the `maxWithdrawable` amount successfully

2. to not make it trivial to raid the highest amount from a card tap without hitting a user set limit

`maxWithdrawable` is useful for the one-time card/QR use case but not in the same way for a reusable card

i.e. I'm happy with `pinLimit` as originally defined in the PR